### ISSUE TITLE
 iquest cannot match names containing apostrophes via = operator if it's not the last condition in the where clause

### ISSUE THREAD

**Issue by lwesterhof** [irods/irods#5727]
- [x] main
- [x] 4-3-stable

### iRODS Version, OS and Version
iRODS v4.2.9
CentOS 7

### What did you try to do?
Query for data objects in collections with an apostrophe. (A special case of  #4887)

### Observed behavior (including steps to reproduce, if applicable)
We have a collection (lazlo's test) with two data objects (lazlo's file and yoda-metadata.json):
```
bash-4.2$ iquest "select DATA_NAME, COLL_NAME where COLL_NAME = '/tempZone/home/research-initial/lazlo's test'"
DATA_NAME = lazlo's file
COLL_NAME = /tempZone/home/research-initial/lazlo's test
------------------------------------------------------------
DATA_NAME = yoda-metadata.json
COLL_NAME = /tempZone/home/research-initial/lazlo's test
------------------------------------------------------------
```

Querying for yoda-metadata.json works fine when the collection name (with apostrophe) is the last condition in the where clause:
```
bash-4.2$ iquest "select DATA_NAME, COLL_NAME where DATA_NAME = 'yoda-metadata.json' AND COLL_NAME = '/tempZone/home/research-initial/lazlo's test'"
DATA_NAME = yoda-metadata.json
COLL_NAME = /tempZone/home/research-initial/lazlo's test
```

The query does not work when the collection name is not the latest condition in the where clause:
```
bash-4.2$ iquest "select DATA_NAME, COLL_NAME where COLL_NAME = '/tempZone/home/research-initial/lazlo's test' AND DATA_NAME = 'yoda-metadata.json'"
CAT_NO_ROWS_FOUND: Nothing was found matching your query

```

Quering for a data object with an apostrophe inside the collection with an apostrophe returns no results:
```
bash-4.2$ iquest "select DATA_NAME, COLL_NAME where DATA_NAME = 'lazlo's file' AND COLL_NAME = '/tempZone/home/research-initial/lazlo's test'"
CAT_NO_ROWS_FOUND: Nothing was found matching your query
bash-4.2$ iquest "select DATA_NAME, COLL_NAME where COLL_NAME = '/tempZone/home/research-initial/lazlo's test' and DATA_NAME = 'lazlo's file'"
CAT_NO_ROWS_FOUND: Nothing was found matching your query
```

**Comment by alanking** [comment]
I ran into this inadvertently via a genquery being used internally for the creation of data objects (#5759).

This will possibly be solved via a new GenQuery parser implementation which has been in the works for some time now, but I'm just noting that this has been reproduced and can lead to some pretty subtle bugs.

**Comment by korydraughn** [comment]
Related #3902.

**Comment by korydraughn** [comment]
This situation is resolved by the GenQuery2 parser (#7570). See the following:
```bash
$ iquery "select DATA_NAME, COLL_NAME where COLL_NAME = '/tempZone/home/rods/lazlo\x27s test' and DATA_NAME = 'lazlo\x27s file'" | jq
[
  [
    "lazlo's file",
    "/tempZone/home/rods/lazlo's test"
  ]
]
```

A test capturing this is all that is needed.
